library/Icingadb/Widget/ViewModeSwitcher.php

```diff
@@ -104,15 +128,28 @@ protected function assemble()
         $currentViewMode = $this->getViewMode();
 
         foreach (static::$viewModes as $viewMode => $icon) {
-            $url = $this->url->with($viewModeParam, $viewMode);
-
-            $link = Html::tag('li', new Link(new IcingaIcon($icon), $url));
+            $input = new HtmlElement('input', [
+                'class' => 'autosubmit',
+                // TODO(el): Protect ID
+                'id'    => $icon,
```

Comment:
> The mode is not persistent through different sessions. I'd expect this to be the case 🤔 You should save the preferences to disk upon submit.

We should discuss this with @lippserd, as this is by design as far as I recall.

Reply:
> * The mode is not persistent through different sessions. I'd expect this to be the case 🤔 You should save the preferences to disk upon submit.

As discussed, we'll leave that out as for having minimum viable solution

> * The switcher now has no need to know the request's url anymore. Please remove the `$url` attribute and all its usages. Then, instead of dropping the shifted value on line 322, populate the form with it: `$viewModeSwitcher->populate(['view' => $this->params->shift($viewModeSwitcher->getViewModeParam())]);`

Not sure, if I'm missing something. But shouldn't there still be the option to control it via url, when the param is explicitly given (e.g. in Dashboard lists)?

Reply:
That's why I told you to not shift and forget the value but pass it to `Form::populate()` ;)

Reply:
Oh, and btw this is the discussion about the id and its protection, not about my general review comments :wink: